src/main_wrapper.c

```diff
@@ -13,288 +13,300 @@
 #include "luagit2/blame/lua_blame.h"
 #include "luagit2/blob/lua_blob.h"
 //----------------- Helper Methods Functions --------------------//
-#include "./luagit2/libgit2/lua_libgit2_help.h"
-#include "./luagit2/signature/lua_signature_help.h"
+#include "luagit2/libgit2/lua_libgit2_help.h"
+#include "luagit2/signature/lua_signature_help.h"
 
 static const struct luaL_Reg luagit2 [] = {
 
       //-------------- libgit2 functions -------------------------------------------------------//
       {"luagit2_init", lua_git_libgit2_init },
-      {"luagit2_shutdown" , lua_git_libgit2_shutdown},
-      {"luagit2_features",lua_git_libgit2_features},
-      {"luagit2_version",lua_git_libgit2_version},
-      {"luagit2_GET_MWINDOW_SIZE",lua_GIT_OPT_GET_MWINDOW_SIZE},
-      {"luagit2_SET_MWINDOW_SIZE",lua_GIT_OPT_SET_MWINDOW_SIZE},
-      {"luagit2_GET_MWINDOW_MAPPED_LIMIT",lua_GIT_OPT_GET_MWINDOW_MAPPED_LIMIT},
-      {"luagit2_SET_MWINDOW_MAPPED_LIMIT",lua_GIT_OPT_SET_MWINDOW_MAPPED_LIMIT},
-      {"luagit2_GET_SEARCH_PATH",lua_GIT_OPT_GET_SEARCH_PATH},
-      {"luagit2_SET_SEARCH_PATH",lua_GIT_OPT_SET_SEARCH_PATH},
-      {"luagit2_GET_CACHED_MEMORY",lua_GIT_OPT_GET_CACHED_MEMORY},
-      {"luagit2_SET_CACHE_OBJECT_LIMIT",lua_GIT_OPT_SET_CACHE_OBJECT_LIMIT},
-      {"luagit2_SET_CACHE_MAX_SIZE",lua_GIT_OPT_SET_CACHE_MAX_SIZE},
-      {"luagit2_ENABLE_CACHING",lua_GIT_OPT_ENABLE_CACHING},
-      {"luagit2_GET_TEMPLATE_PATH",lua_GIT_OPT_GET_TEMPLATE_PATH},
-      {"luagit2_SET_TEMPLATE_PATH",lua_GIT_OPT_SET_TEMPLATE_PATH},
-      {"luagit2_SET_USER_AGENT",lua_GIT_OPT_SET_USER_AGENT},
-      {"luagit2_GET_WINDOWS_SHAREMODE",lua_GIT_OPT_GET_WINDOWS_SHAREMODE},
-      {"luagit2_SET_WINDOWS_SHAREMODE",lua_GIT_OPT_SET_WINDOWS_SHAREMODE},
-      {"luagit2_ENABLE_STRICT_OBJECT_CREATION",lua_GIT_OPT_ENABLE_STRICT_OBJECT_CREATION},
-      {"luagit2_ENABLE_STRICT_SYMBOLIC_REF_CREATION",lua_GIT_OPT_ENABLE_STRICT_SYMBOLIC_REF_CREATION},
-      {"luagit2_SET_SSL_CIPHERS",lua_GIT_OPT_SET_SSL_CIPHERS},
-      {"luagit2_ENABLE_OFS_DELTA",lua_GIT_OPT_ENABLE_OFS_DELTA},
-      {"luagit2_ENABLE_FSYNC_GITDIR",lua_GIT_OPT_ENABLE_FSYNC_GITDIR},
-      {"luagit2_ENABLE_STRICT_HASH_VERIFICATION",lua_GIT_OPT_ENABLE_STRICT_HASH_VERIFICATION},
+      {"luagit2_shutdown", lua_git_libgit2_shutdown},
+      {"luagit2_features", lua_git_libgit2_features},
+      {"luagit2_version", lua_git_libgit2_version},
```

Comment:
The naming of the functions are quite confusing. Here, for example "luagit2_version" could mean that you are getting the version of the lua library, when in fact it refers to the libgit2 original library. 

What I would suggest is not to use the `luagit2_` prefix in any of the namings, and here for the libgit2 group, have the `libgit2_` prefix. That means that in the future, calling these functions would look like this:

```
local luagit2 = require "luagit2"
luagit2.libgit2_version();
luagit2.repository_init();
```

instead of 

```
local lib = require "luagit2"
lib.luagit2_version();
lib.luagit2_repository_init();
```

In this way we adhere to the best Lua practices of namespacing the libraries, and we dont mess the libgit2 original namespacing :)

Reply:
Alright I will do so.